serving/src/test/java/feast/serving/it/ServingServiceBigTableIT.java

```diff
@@ -0,0 +1,526 @@
+import feast.proto.serving.ServingServiceGrpc;
+import feast.proto.types.ValueProto;
+import io.grpc.ManagedChannel;
+import io.grpc.ManagedChannelBuilder;
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.avro.Schema;
+import org.apache.avro.SchemaBuilder;
+import org.apache.avro.generic.GenericDatumWriter;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.generic.GenericRecordBuilder;
+import org.apache.avro.io.*;
+import org.junit.ClassRule;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.context.annotation.Bean;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.containers.DockerComposeContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@ActiveProfiles("it")
+@SpringBootTest(
+    webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+    properties = {
+      "feast.core-cache-refresh-interval=1",
+      "feast.active_store=bigtable",
+      "spring.main.allow-bean-definition-overriding=true"
+    })
+@Testcontainers
+public class ServingServiceBigTableIT extends BaseAuthIT {
+
+  static final Map<String, String> options = new HashMap<>();
+  static final String timestampPrefix = "_ts";
+  static CoreSimpleAPIClient coreClient;
+  static ServingServiceGrpc.ServingServiceBlockingStub servingStub;
+
+  static final int FEAST_SERVING_PORT = 6569;
+
+  static final String PROJECT_ID = "test-project";
+  static final String INSTANCE_ID = "test-instance";
+  static ManagedChannel channel;
+
+  @ClassRule @Container
+  public static DockerComposeContainer environment =
+      new DockerComposeContainer(
+              new File("src/test/resources/docker-compose/docker-compose-bigtable-it.yml"))
+          .withExposedService(
+              CORE,
+              FEAST_CORE_PORT,
+              Wait.forLogMessage(".*gRPC Server started.*\\n", 1)
+                  .withStartupTimeout(Duration.ofMinutes(SERVICE_START_MAX_WAIT_TIME_IN_MINUTES)))
+          .withExposedService(BIGTABLE, BIGTABLE_PORT);
+
+  @DynamicPropertySource
+  static void initialize(DynamicPropertyRegistry registry) {
+    registry.add("grpc.server.port", () -> FEAST_SERVING_PORT);
+  }
+
+  @BeforeAll
+  static void globalSetup() throws IOException {
+    coreClient = TestUtils.getApiClientForCore(FEAST_CORE_PORT);
```

Comment:
this method has 200 loc. it's simply hard to read. can we split it into functions and maybe eliminate some common parts?

Reply:
And should we probably move data creation into relevant tests? instead of creating them all together.

Reply:
Yeap, it would be cleaner by refactoring common parts out. As for data creation, I think it's cleaner to isolate data generation before all tests start running and only keep retrieval logic in the tests.